gnat2goto/driver/range_check.ads

```diff
@@ -1,16 +1,23 @@
 with Uintp;                     use Uintp;
 with Urealp;                    use Urealp;
 with Ada.Containers.Vectors;    use Ada.Containers;
+with Ada.Containers.Ordered_Maps;
 
 with Ireps;                     use Ireps;
 with Types;                     use Types;
 
 package Range_Check is
 
+   type Bound_Type is (Nat_Bound, Real_Bound, Symb_Bound);
+
    type Bound_Type_Nat is new Uint;
    type Bound_Type_Real is new Ureal;
    type Bound_Type_Expr is new Irep;
 
+   package All_Bounds_Map is new
+     Ada.Containers.Ordered_Maps (Key_Type     => Natural,
+                                  Element_Type => Bound_Type);
```

Comment:
This does not in fact guarantee uniqueness of ids for individual bounds: `All_Bounds` should be a vector and the `Nat,Real,Expr_Bounds` should be maps. I kept it here to ease reviewing. It is addressed in the last commit.